kinoml/features/complexes.py

```diff
@@ -0,0 +1,544 @@
+        """
+        Perform docking with OpenEye toolkits and thoughtful defaults.
+        Parameters
+        ----------
+        system: ProteinLigandComplex
+            A system object holding protein and ligand information.
+        Returns
+        -------
+        protein_ligand_complex: ProteinLigandComplex
+            The same system but with docked ligand.
+        """
+        ligands = self._read_ligands(system.ligand.path)
+        protein = self._read_protein(system.protein.path)
+
+        # TODO: more sophisticated decision between hybrid and chemgauss docking
+        if self._has_ligand(protein):
+
+            # TODO: electron density, loop database
+            prepared_protein, prepared_ligand = self._prepare_complex(protein)
+            hybrid_receptor = self._create_hybrid_receptor(
+                prepared_protein, prepared_ligand
+            )
+            docking_poses = self._hybrid_docking(hybrid_receptor, ligands)
+
+        else:
+            # TODO: identify pocket with point or box
+            # TODO: check possibility to define design unit with residue (would consider electron density)
+            raise NotImplemented
+
+        # TODO: where to store data
+        protein_path = "/path/to/protein.pdb"  # mmcif writing not supported by openeye
+        self._write_molecules([prepared_protein], protein_path)
+        file_protein = FileProtein(path=protein_path)
+
+        ligand_path = "/path/to/ligand.sdf"
+        self._write_molecules(docking_poses, ligand_path)
+        file_ligand = FileLigand(path=ligand_path)
+        protein_ligand_complex = ProteinLigandComplex(
+            components=[file_protein, file_ligand]
+        )
+
+        return protein_ligand_complex
+
+    @staticmethod
+    def _read_ligands(path: str) -> List[oechem.OEGraphMol]:
+        """
+        Read ligands from a file.
+        Parameters
+        ----------
+        path: str
+            Path to ligand file.
+        Returns
+        -------
+        molecules: list of oechem.OEGraphMol
+            A List of ligands as OpenEye molecules.
+        """
+        from openeye import oechem
+
+        molecules = []
+        with oechem.oemolistream(path) as ifs:
+            for molecule in ifs.GetOEGraphMols():
+                molecules.append(oechem.OEGraphMol(molecule))
+
+        # TODO: returns empty list if something goes wrong
+        return molecules
+
+    @staticmethod
+    def _read_protein(path: str) -> oechem.OEGraphMol:
+        """
+        Read a protein from a file.
+        Parameters
+        ----------
+        path: str
+            Path to protein file.
+        Returns
+        -------
+        molecule: oechem.OEGraphMol
+            A protein as OpenEye molecule.
+        """
+        from openeye import oechem
+
+        suffix = path.split(".")[-1]
+        with oechem.oemolistream(path) as ifs:
+            if suffix == "pdb":
+                ifs.SetFlavor(
```

Comment:
This was taken from [here](https://docs.eyesopen.com/toolkits/python/oechemtk/molreadwrite.html#flavored-input-and-output). I guess we need this to combine different possible `PDB` formats we come across?